src/components/SlateEditor/plugins/concept/EditSlateConcept.tsx

```diff
@@ -25,78 +29,91 @@ const getConceptDataAttributes = ({ id, title: { title } }) => ({
   },
 });
 
-const EditSlateConcept = props => {
-  const { t, children, node, locale, editor, attributes } = props;
-  const nodeText = node.text.trim();
+interface Props {
+  element: ConceptElement;
+  locale: string;
+  editor: Editor;
+  attributes: {
+    'data-key': string;
+  };
+  children: ReactNode;
+}
+
+const EditSlateConcept = (props: Props & tType) => {
+  const { t, children, element, locale, editor, attributes } = props;
+  const nodeText = Node.string(element).trim();
 
   const [showConcept, setShowConcept] = useState(false);
 
-  const toggleConceptModal = evt => {
+  const toggleConceptModal = (evt?: React.MouseEvent) => {
     if (evt) {
       evt.preventDefault();
     }
     setShowConcept(!showConcept);
   };
 
   const { concept, subjects, ...conceptHooks } = useFetchConceptData(
-    node.data.get('content-id'),
+    parseInt(element.data['content-id']),
     locale,
   );
   const conceptId = concept && concept.id ? concept.id : undefined;
 
-  const handleChangeAndClose = editor => {
-    editor.focus(); // Always return focus to editor
+  const handleChangeAndClose = (editor: Editor) => {
+    ReactEditor.focus(editor); // Always return focus to editor
     toggleConceptModal();
   };
 
-  const addConcept = addedConcept => {
+  const addConcept = (addedConcept: ConceptFormType) => {
     const data = getConceptDataAttributes({
       ...addedConcept,
       title: { title: nodeText },
     });
-    if (node.key) {
-      handleChangeAndClose(editor.moveToRangeOfNode(node).setInlines(data));
+    if (element) {
+      const path = ReactEditor.findPath(editor, element);
+      Transforms.setNodes(
+        editor,
+        { data: data.data },
+        { at: path, match: node => Element.isElement(node) && node.type === TYPE_CONCEPT },
+      );
+      mergeLastUndos(editor);
+      handleChangeAndClose(editor);
     }
   };
 
   const handleRemove = () => {
-    const nextValue = editor
-      .moveToRangeOfNode(node)
-      .removeNodeByKey(node.key)
-      .insertText(node.text);
-    handleChangeAndClose(nextValue);
+    const path = ReactEditor.findPath(editor, element);
+
+    Transforms.removeNodes(editor, {
+      at: path,
+      match: node => Element.isElement(node) && node.type === TYPE_CONCEPT,
+    });
+    handleChangeAndClose(editor);
```

Comment:
For å holde oppførsel lik som det er fra før må det settes inn en paragraf med teksten til elementet, evt kan du muligens bare bruke transform å gjøre det fra concept til paragraf.

Reply:
Brukte unwrapNodes. Concept er en inline-node, så da blir bare teksten som er inni værende og concept-wrapperen forsvinner.